js/actionCreators/files.js

```diff
@@ -236,14 +229,35 @@ export function fetchMediaTags(file, id) {
       .then(data => {
         // There's more data here, but we don't have a use for it yet:
         // https://github.com/aadsm/jsmediatags#shortcuts
-        const { artist, title, picture } = data.tags;
+        const { artist, title, picture } = data.common;
         let albumArtUrl = null;
-        if (picture) {
-          const byteArray = new Uint8Array(picture.data);
-          const blob = new Blob([byteArray], { type: picture.type });
+        if (picture && picture.length >= 1) {
+          const byteArray = new Uint8Array(picture[0].data);
+          const blob = new Blob([byteArray], { type: picture[0].format });
           albumArtUrl = URL.createObjectURL(blob);
         }
         dispatch({ type: SET_MEDIA_TAGS, artist, title, albumArtUrl, id });
+
+        // If we're not going to load this right away,
+        // we should set duration on our own
+        if (data.format.duration != null) {
+          dispatch({
+            type: SET_MEDIA_DURATION,
+            duration: data.format.duration,
+            id
+          });
+        } else {
+          //  ToDo? dispatch(fetchMediaDuration(canonicalUrl, id));
+        }
+
+        dispatch({
+          type: SET_MEDIA,
+          kbps: Math.round(data.format.bitrate / 1000).toString(),
+          khz: Math.round(data.format.sampleRate / 1000).toString(),
+          channels: data.format.channels,
```

Comment:
Didn't even think of this! This was another piece of data that I wasn't able to get from the browser. 👏